hazelcast-jet-core/src/test/java/com/hazelcast/jet/Processors_peekTest.java

```diff
@@ -0,0 +1,184 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet;
+
+import com.hazelcast.jet.function.DistributedFunction;
+import com.hazelcast.jet.function.DistributedPredicate;
+import com.hazelcast.jet.function.DistributedSupplier;
+import com.hazelcast.jet.impl.execution.init.Contexts.ProcCtx;
+import com.hazelcast.jet.impl.util.ArrayDequeInbox;
+import com.hazelcast.jet.impl.util.ArrayDequeOutbox;
+import com.hazelcast.jet.impl.util.ProgressTracker;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.nio.Address;
+import com.hazelcast.test.HazelcastParametersRunnerFactory;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import static com.hazelcast.jet.Processors.peekInput;
+import static com.hazelcast.jet.Processors.peekOutput;
+import static com.hazelcast.jet.function.DistributedFunctions.alwaysTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+
+@RunWith(Parameterized.class)
+@Category({QuickTest.class, ParallelTest.class})
+@Parameterized.UseParametersRunnerFactory(HazelcastParametersRunnerFactory.class)
+public class Processors_peekTest {
+
+    private ArrayDequeInbox inbox;
+    private ArrayDequeOutbox outbox;
+    private Processor.Context context;
+    private ILogger logger;
+
+    @Parameter
+    public DistributedFunction<Object, String> toStringF;
+
+    @Parameter(1)
+    public DistributedPredicate<Object> shouldLogF;
+
+    @Parameters(name = "toStringF={0}, shouldLogF={1}")
+    public static Collection<Object[]> parameters() {
+        return Arrays.asList(
+                new Object[]{null, null},
+                new Object[]{
+                        (DistributedFunction<Object, String>) o -> "a" + o,
+                        (DistributedPredicate<Object>) o -> (int) o <= 1
+                }
+        );
+    }
+
+    @Before
+    public void before() {
+        inbox = new ArrayDequeInbox();
+        outbox = new ArrayDequeOutbox(new int[]{128}, new ProgressTracker());
+        logger = mock(ILogger.class);
+        context = new ProcCtx(null, logger, null, 0);
+    }
+
+    @Test
+    public void when_peekInput_SupplierProcessor() {
+        // Given
+        DistributedSupplier<Processor> passThroughPSupplier = Processors.filter(alwaysTrue());
+        Processor wrappedP =
+                (toStringF == null ? peekInput(passThroughPSupplier) : peekInput(toStringF, shouldLogF, passThroughPSupplier))
```

Comment:
Line length exceeds 120 here and in several more places below. This rule should be present in CheckStyle, though. I'll add it.